### PR TITLE
goroutine: Improve unit test

### PR DESCRIPTION
**Pull request by efritz**
Make this thing not suck.

## Test plan

...the unit test

### CODE REVIEW

File: internal/goroutine/periodic_test.go [hunk 1]
```diff
@@ -196,37 +200,23 @@ func TestPeriodicGoroutineWithDynamicConcurrency(t *testing.T) {
...
-	clock.BlockingAdvance(time.Second)
-	<-called
-	clock.BlockingAdvance(time.Second)
-	<-called
-	<-called
-
-	// Pool size 3
-	concurrencyClock.BlockingAdvance(concurrencyRecheckInterval)
+	for poolSize := 1; poolSize < 3; poolSize++ {
+		// Ensure each of the
```
Comment (camdencheek) [inline internal/goroutine/periodic_test.go:204]: ?

Comment (efritz) [reply internal/goroutine/periodic_test.go:204]: 🆗

**Review by camdencheek — APPROVED** [review]

File: internal/goroutine/periodic_test.go (hunk 1 above)
Comment (efritz) [reply internal/goroutine/periodic_test.go:204]: DAMN IT AUTO MERGE

Comment (efritz) [reply internal/goroutine/periodic_test.go:204]: https://github.com/sourcegraph/sourcegraph/pull/53033

**Pull request merged**